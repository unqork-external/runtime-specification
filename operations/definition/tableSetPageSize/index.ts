export * from './tableSetPageSize.model'
export * from './tableSetPageSize.options'
