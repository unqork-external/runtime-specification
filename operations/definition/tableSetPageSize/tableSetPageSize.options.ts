import { Any, Description, Property } from '@tsed/schema'

export class TableSetPageSizeOperationOptions {
  @Description('The key for the component we want to set page size')
  @Property()
  targetKey: string

  @Description('Page size')
  @Any('number', 'string')
  pageSize: number | string
}
