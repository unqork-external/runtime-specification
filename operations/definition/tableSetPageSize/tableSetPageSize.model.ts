import { Description, DiscriminatorValue, Required } from '@tsed/schema'

import { TableSetPageSizeOperationOptions } from './tableSetPageSize.options'
import { Stable } from '../../../../decorators/stability/stable.decorator'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@Description('Sets the number of rows displayed per page of the Table component.')
@Stable()
@DiscriminatorValue(OperationTypes.TABLE_SET_PAGE_SIZE)
export class TableSetPageSizeOperation extends Operation<OperationTypes.TABLE_SET_PAGE_SIZE> {
  public type = OperationTypes.TABLE_SET_PAGE_SIZE as const

  @Required()
  public options: TableSetPageSizeOperationOptions = new TableSetPageSizeOperationOptions()
}
