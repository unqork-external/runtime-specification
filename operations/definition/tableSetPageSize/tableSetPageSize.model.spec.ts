import { TableSetPageSizeOperation } from './tableSetPageSize.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: TableSetPageSize Tests', () => {
  let validate, schema
  beforeAll(() => {
    const response = generateSchemaAndValidate(TableSetPageSizeOperation)
    validate = response.validate
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema).toMatchSnapshot()
  })
})
