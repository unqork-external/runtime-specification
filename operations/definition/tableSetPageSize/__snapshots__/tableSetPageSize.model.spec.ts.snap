// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: TableSetPageSize Tests should match this JSON Schema 1`] = `
{
  "definitions": {
    "TableSetPageSizeOperationOptions": {
      "properties": {
        "pageSize": {
          "description": "Page size",
          "type": [
            "number",
            "string",
          ],
        },
        "targetKey": {
          "description": "The key for the component we want to set page size",
          "type": "string",
        },
      },
      "type": "object",
    },
  },
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/TableSetPageSizeOperationOptions",
    },
    "type": {
      "const": "TABLE_SET_PAGE_SIZE",
      "examples": [
        "TABLE_SET_PAGE_SIZE",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "type": "object",
}
`;
